docs/docs/writing-plugins/the-rules-api/concepts.mdx

```diff
@@ -23,176 +23,251 @@ async def int_to_str(i: int) -> str:
 
-In this example, we could not have requested the type `ProcessResult` as a parameter to our rule because we needed to dynamically create a `Process` object.
+### Extra context for implicit parameters
 
-Thanks to `await Get`, we can write a recursive rule to compute a [Fibonacci number](https://en.wikipedia.org/wiki/Fibonacci_number):
+As mentioned above, Pants can compute values for implicit parameters by transitively applying rules. In many cases the initial parameters for *those* rules are known from external context. But in some cases we need to provide extra context from the calling rule. To do so, we pass the contextual parameters as arguments to `**implicitly()`:
 
 ```python
-@dataclass(frozen=True)
-class Fibonacci:
-    val: int
+from pants.engine.process import fallible_to_exec_result_or_raise
+from pants.engine.rules import implicitly, rule
 
 @rule
-async def compute_fibonacci(n: int) -> Fibonacci:
-    if n < 2:
-        return Fibonacci(n)
-    x = await Get(Fibonacci, int, n - 2)
-    y = await Get(Fibonacci, int, n - 1)
-    return Fibonacci(x.val + y.val)
+async def my_rule() -> MyResult:
+    process_result = await fallible_to_exec_result_or_raise(
+        **implicitly(
+            Process(
+                ["/bin/echo", str(target.address)],
+                description=f"Echo {target.address}",
+            )
+        )
+    )
+    ...
 ```
 
-Another rule could then "call" our Fibonacci rule by using its own `Get`:
+In this example the `fallible_to_exec_result_or_raise()` rule takes a `FallibleProcessResult` and returns a `ProcessResult` by first checking the `FallibleProcessResult` for success and raising an exception if it failed. We saw this earlier, in the simplified shellcheck example.
 
-```python
-@rule
-async def call_fibonacci(...) -> Foo:
-    fib = await Get(Fibonnaci, int, 4)
-    ...
-```
+But instead of explicitly passing a `FallibleProcessResult` as we did earlier, we now pass a `Process` as implicit context. The Pants engine then looks at all the rules it knows about to figure out how to compute a `FallibleProcessResult` from a `Process`. The `execute_process()` we encountered earlier fits the bill, and so the engine calls it on our `Process` and passes its return value into `fallible_to_exec_result_or_raise()`. Whereas earlier we called both rules explicitly, here we get the exact same behavior with just one call.
 
-:::note `Get` constructor shorthand
-The verbose constructor for a `Get` object takes three parameters: `Get(OutputType, InputType, input)`, where `OutputType` and `InputType` are both types, and `input` is an instance of `InputType`.
+In fact, since raising an exception on process failure is frequently what you want, we have an alias, `execute_process_or_raise`, to make the code more readable when using this common shorthand idiom.
```

Comment:
I don't think this is necessary info, just ditch the old, verbose version - and we can clean up the Pants code to use `execute_process_or_raise` exclusively.